File: CSharp/PolyData/IterateOverLines.cs
using System;
using Kitware.VTK;

namespace ActiViz.Examples {
   class Program {
      static void Main(string[] args) {
         IterateOverLines();
         Console.WriteLine("\nPress any key to continue...");
         Console.ReadKey();
      }


      private static void IterateOverLines() {
         double[] origin = new double[] { 0.0, 0.0, 0.0 };
         double[,] p = new double[,] {
            {1.0, 0.0, 0.0},
            {0.0, 1.0, 0.0},
            {0.0, 1.0, 2.0},
            {1.0, 2.0, 3.0}};

         // Create a vtkPoints object and store the points in it
         vtkPoints points = vtkPoints.New();
         points.InsertNextPoint(origin[0], origin[1], origin[2]);
         for(int i = 0; i < 4; i++)
            points.InsertNextPoint(p[i, 0], p[i, 1], p[i, 2]);

         // Create a cell array to store the lines in and add the lines to it
         vtkCellArray lines = vtkCellArray.New();

         // Create four lines
         for(int i = 0; i < 4; i++) {
            vtkLine line = vtkLine.New();
            line.GetPointIds().SetId(0, i);
            line.GetPointIds().SetId(1, i + 1);
            lines.InsertNextCell(line);
         }

         // Create a polydata to store everything in
         vtkPolyData linesPolyData = vtkPolyData.New();

         // Add the points to the dataset
         linesPolyData.SetPoints(points);

         // Add the lines to the dataset
         linesPolyData.SetLines(lines);

         Console.WriteLine("There are " + linesPolyData.GetNumberOfLines() + " lines.");
         linesPolyData.GetLines().InitTraversal();
         vtkIdList idList = vtkIdList.New();
         while(linesPolyData.GetLines().GetNextCell(idList) != 0) {
            Console.WriteLine("Line has " + idList.GetNumberOfIds() + " points.");

            for(int pointId = 0; pointId < idList.GetNumberOfIds(); pointId++) {
               Console.Write(idList.GetId(pointId) + " ");
            }
            Console.Write(Environment.NewLine);
         }
      }
   }
}

File: src/Python/PolyData/IterateOverLines.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import vtk

def main():
    origin = [0.0, 0.0, 0.0]
    p0 = [1.0, 0.0, 0.0]
    p1 = [0.0, 1.0, 0.0]
    p2 = [0.0, 1.0, 2.0]
    p3 = [1.0, 2.0, 3.0]

    # Create a vtkPoints object and store the points in it.
    points = vtk.vtkPoints()
    points.InsertNextPoint(origin)
    points.InsertNextPoint(p0)
    points.InsertNextPoint(p1)
    points.InsertNextPoint(p2)
    points.InsertNextPoint(p3)

    # Create a cell array to store the lines in and add the lines to it.
    lines = vtk.vtkCellArray()

    # Create four lines.
    for i in range(4):
        line = vtk.vtkLine()
        line.GetPointIds().SetId(0, i)
        line.GetPointIds().SetId(1, i + 1)
        lines.InsertNextCell(line)

    # Create a polydata to store everything in.
    linesPolyData = vtk.vtkPolyData()

    # Add the points to the dataset.
    linesPolyData.SetPoints(points)

    # Add the lines to the dataset.
    linesPolyData.SetLines(lines)

    print("There are {0} lines.".format(linesPolyData.GetNumberOfLines()))

    linesPolyData.GetLines().InitTraversal()
    idList = vtk.vtkIdList()
    while(linesPolyData.GetLines().GetNextCell(idList)):
        print("Line has {0} points".format(idList.GetNumberOfIds()))
        for pointId in range(idList.GetNumberOfIds() - 1):
            print("{0} {1}".format(idList.GetId(pointId), idList.GetId(pointId + 1)))

if __name__ == "__main__":
    main()
